File: programming_paradigms/object_oriented_programming/java_examples/ObjectOrientedIntermediate.java
package programming_paradigms.object_oriented_programming.java_examples;

class Student {
    private String name;
    private int age;
    private String studentId;

    public Student(String name, int age, String studentId) {
        this.name = name;
        this.age = age;
        this.studentId = studentId;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getStudentId() {
        return studentId;
    }

    public void displayInfo() {
        System.out.println("Name: " + name + ", Age: " + age + ", Student ID: " + studentId);
    }
}

public class ObjectOrientedIntermediate {
    public static void main(String[] args) {
        Student student = new Student("Alice", 20, "12345");
        student.displayInfo();
    }
}

File: programming_paradigms/object_oriented_programming/golang_examples/ObjectOrientedIntermediate.go
package golangexamples

import "fmt"

type Student struct {
	Name      string
	Age       int
	StudentID string
}

func (s Student) DisplayInfo() {
	fmt.Printf("Name: %s, Age: %d, Student ID: %s\n", s.Name, s.Age, s.StudentID)
}

func main() {
	student := Student{Name: "Alice", Age: 20, StudentID: "12345"}
	student.DisplayInfo()
}
